medvis/dicom/src/io/gdcmvolumereader.cpp

```diff
@@ -0,0 +1,1070 @@
+
+    // Photometric interpretation string
+    // e.g. MONOCHROME1 where lowest value is white or MONOCHROME2 where lowest value is black
+    // https://dicom.innolitics.com/ciods/mr-image/image-pixel/00280004
+    std::string piString;
+
+    if (imageStack.size() > 0) {
+        const gdcm::Image& image =
+            imageStack[0];  // just assume that all images have consistent values
+
+        auto colorspace = image.GetPhotometricInterpretation();
+        colorspace.GetString();
+        channelCount = colorspace.GetSamplesPerPixel();
+
+        origin.x = image.GetOrigin(0);
+        origin.y = image.GetOrigin(1);
+        origin.z = image.GetOrigin(2);
+
+        spacing.x = image.GetSpacing(0);
+        spacing.y = image.GetSpacing(1);
+
+        if (sliceThickness == std::numeric_limits<double>::infinity()) {
+            spacing.z = image.GetSpacing(2);  // fallback if no explicit slice thickness is set
+        } else {
+            spacing.z =
+                sliceThickness;  // image.GetSpacing(2); seems to be not set correctly for volumes?
+        }
+
+        pixelformat = image.GetPixelFormat();
+
+        rescaleIntercept = image.GetIntercept();
+        rescaleSlope = image.GetSlope();
+
+        channelBits = image.GetPixelFormat().GetBitsAllocated();
+        usedChannelBits = image.GetPixelFormat().GetBitsStored();
+        isSignedInt = image.GetPixelFormat().GetPixelRepresentation();
+    } else {
+        return 0;  // empty volume
+    }
+
+    // Save info that is needed during GCDMVolumeRAMLoader::dispatch
+    series.slope = rescaleSlope;
+    series.intercept = rescaleIntercept;
+
+    // Now map metadata to inviwo data types
+
+    // The volume should have enough space for the largest image
+    series.dims = size3_t(maxWidth, maxHeight, imageStack.size());
+
+    const auto voxelFormat =
+        DataFormatBase::get(isSignedInt ? NumericType::SignedInteger : NumericType::UnsignedInteger,
+                            channelCount, channelBits);
+
+    const SharedVolume outputVolume = std::make_shared<Volume>(series.dims, voxelFormat);
+
+    // Compute data range from the used bits, differentiating between unsigned and 2's complement
+    // or...
+    outputVolume->dataMap_.dataRange =
+        isSignedInt ? dvec2{-std::pow(2, usedChannelBits - 1), std::pow(2, usedChannelBits - 1) - 1}
+                    : dvec2{0, std::pow(2, usedChannelBits) - 1};
+
+    // ... use DICOM's explicit datarange if existing
+    bool explicitRange = false;
+    if (smallestVoxelValue < std::numeric_limits<double>::infinity()) {
+        outputVolume->dataMap_.dataRange.x = smallestVoxelValue;
+        explicitRange = true;
+    }
+    if (largestVoxelValue > -std::numeric_limits<double>::infinity()) {
+        outputVolume->dataMap_.dataRange.y = largestVoxelValue;
+        explicitRange = true;
+    }
+
+    // Convert disk data to hounsfield value using the DICOM rescale tags (this is only
+    // valid for CT scans)
```

Comment:
Not sure if this is only valid for ct...

Reply:
Original comment by @mskr, @VisComKreiser ...

Reply:
@ResearchDaniel You are right, we should remove the part in paranthesis.

If I remember correctly, this is just a leftover, since we assumed CT data previously.